File: app/src/main/java/com/deadcells/modding/PakTool.java
package com.deadcells.modding;

public class PakTool {
    static {
        System.loadLibrary("asset_replacer");
    }

    public static native boolean unpack(String pakPath, String outputDir);
    public static native boolean pack(String inputDir, String outputPak, String stamp);
    public static native boolean merge(String outputPak, String stamp, String[] inputPaks);
    public static native boolean atlasUnpack(String atlasPath, String outputDir);
    public static native boolean atlasPack(String inputDir, String outputAtlas, String outputPng);
}

File: app/src/main/java/com/deadcells/modding/PakTool.kt
package com.deadcells.modding

object PakTool {
    init {
        System.loadLibrary("asset_replacer")
    }

    external fun unpack(pakPath: String, outputDir: String): Boolean
    external fun pack(inputDir: String, outputPak: String, stamp: String?): Boolean
    external fun merge(outputPak: String, stamp: String?, vararg inputPaks: String): Boolean
    external fun atlasUnpack(atlasPath: String, outputDir: String): Boolean
    external fun atlasPack(inputDir: String, outputAtlas: String, outputPng: String): Boolean
}
